rest/route/project.go

```diff
@@ -748,3 +749,56 @@ func (h *getProjectVersionsHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return resp
 }
+
+type GetProjectAliasResultsHandler struct {
+	sc data.Connector
+
+	version             string
+	alias               string
+	includeDependencies bool
+}
+
+func makeGetProjectAliasResultsHandler(sc data.Connector) gimlet.RouteHandler {
+	return &GetProjectAliasResultsHandler{
+		sc: sc,
+	}
+}
+
+func (h *GetProjectAliasResultsHandler) Factory() gimlet.RouteHandler {
+	return &GetProjectAliasResultsHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *GetProjectAliasResultsHandler) Parse(ctx context.Context, r *http.Request) error {
+	params := r.URL.Query()
+
+	h.version = params.Get("version")
+	if h.version == "" {
+		return errors.New("'version' parameter must be specified")
+	}
+	h.alias = params.Get("alias")
+	if h.alias == "" {
+		return errors.New("'alias' parameter must be specified")
+	}
+	h.includeDependencies = (params.Get("include_deps") == "true")
+
+	return nil
+}
+
+func (h *GetProjectAliasResultsHandler) Run(ctx context.Context) gimlet.Responder {
+	proj, err := dbModel.FindProjectFromVersionID(h.version)
```

Comment:
This should still use h.sc though and not explicitly dbModel

Reply:
I don't see a data method that's sufficiently close. Do you have one in mind?

Reply:
Well what I mean is you should make one, similarly to how you had to make GetProjectAliasResults to return DB results to an API route.

Reply:
That'll need to take in a version and return a db model though, which feels like it's entirely db logic to me and doesn't belong there

Reply:
I've had a new thought! GetProjectAliasResults should probably then take in a version (like GetProjectAliasResultsForVersion) and call FindProjectFromVersionID in there--there's no reason to use this function explicitly in the route.

Reply:
I had considered that, but it actually makes the interface messier if we want to also support passing in the project config directly. With the interface like this, the route parse method can retrieve the project if it needs, and just pass it in here

Reply:
I think if we haven't decided to pass in the project config yet then we can create a different handler / refactor a bit when the time comes, but this seems better than returning a full project to the route when it doesn't actually need the project.

Reply:
Also I believe in rules i.e. the data connector 😅  that being said I'm not going to keep blocking your merge on this